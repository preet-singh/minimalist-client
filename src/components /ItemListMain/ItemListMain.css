
.ItemListMain ul {
    list-style-type: none;
    margin: 0;
    padding: 10px;
  }
  
  .ItemListMain_button-container {
    max-width: 380px;
    margin: 0 10px;
  }
  
  .ItemListMain_add-note-button {
    margin: 0 auto 20px;
  }
  .ItemListMain_add-note-button:hover {
    background-color: #29862e;
    color: white;
  }
  