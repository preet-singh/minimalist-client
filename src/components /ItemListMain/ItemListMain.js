import React from 'react'
import { Link } from 'react-router-dom'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Item from '../Item/Item'
import Button from '../Button/Button'
import InventoryContext from '../../InventoryContext'
import { getItemsForInventory } from '../item-helpers'
import './ItemListMain.css'
import PropTypes from 'prop-types';

export default class ItemListMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = InventoryContext;

  render() {
    const { inventoryId } = this.props.match.params
    const { items=[] } = this.context
    const itemsForInventory = getItemsForInventory(items, inventoryId)
    return (
      <section className='ItemListMain'>
        <ul>
          {itemsForInventory.map(item =>
            <li key={Number(item.id)}>
              <Item
                id={item.id}
                name={item.item_name}
                modified={Date(item.date_modified).toString()}
              />
            </li>
          )}
        </ul>
        <div className='ItemListMain_button-container'>
          <Button
            tag={Link}
            to='/add-item'
            type='button'
            className='ItemListMain_add-item-button'
          >
            <FontAwesomeIcon icon='plus' />
            <br />
            Item
          </Button>
        </div>
      </section>
    )
  }
}

ItemListMain.propTypes = {
  match: PropTypes.object.isRequired
}
