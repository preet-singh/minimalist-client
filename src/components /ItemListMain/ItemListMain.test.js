import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemListMain from './ItemListMain'

describe(`ItemListMain component`, () => {
  it('renders a .ItemListMain by default', () => {
    const wrapper = shallow(<ItemListMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a Item in ul for each items in array', () => {
    const props = {
      match: {
        params: {
          inventoryId: 'THIS_INVEN_ID'
        }
      }
    }
    const context = {
      items: [
        {
          "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Kitchen",
          "inventoryId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Corporis accusamus placeat.\n \rUnde."
        },
        {
          "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Clothing",
          "inventoryId": "THIS_INVEN_ID",
          "content": "Eos\n \rlaudantium."
        },
        {
          "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Bedding",
          "inventoryId": "THIS_FOLDER_ID",
          "content": "Occaecati dignissimos\nvoluptatum nihil."
        },
        {
          "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Birds",
          "inventoryId": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
          "content": "Eum culpa odit."
        },
      ]
    }
    const ul = shallow(<ItemListMain {...props} />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
