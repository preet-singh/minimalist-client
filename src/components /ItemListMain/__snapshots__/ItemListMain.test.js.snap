// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListMain component renders a .ItemListMain by default 1`] = `
<section
  className="ItemListMain"
>
  <Route
    path="/"
  />
  <ul />
  <div
    className="ItemListMain_button-container"
  >
    <NavButton
      className="ItemListMain_add-item-button"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/add-item"
      type="button"
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={false}
        flip={null}
        icon="plus"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <br />
      Item
    </NavButton>
  </div>
</section>
`;

exports[`ItemListMain component renders a Item in ul for each items in array 1`] = `<ul />`;
